apps/emails-and-messages/src/modules/smtp/configuration/smtp-configuration.router.ts

```diff
@@ -29,8 +62,14 @@ const protectedWithConfigurationService = protectedClientProcedure.use(({ next,
     ctx: {
       ...ctx,
       configurationService: new SmtpConfigurationService({
-        apiClient: ctx.apiClient,
-        saleorApiUrl: ctx.saleorApiUrl,
+        metadataManager: new SmtpPrivateMetadataManager(
+          createSettingsManager(ctx.apiClient, {
+            appId: ctx.appId!,
+            saleorApiUrl: ctx.saleorApiUrl,
+            token: ctx.token!,
+          }),
+          ctx.saleorApiUrl
+        ),
```

Comment:
You add to every method `configurationService` but its actually  `SmtpConfigurationService`

This could have work if you have sort of dependency injection here and you dynamically inject manager for SMTP or Sendgrid, depending on routes. But this would be overengineered.

Here you name something specific like something generic, so someone later will call `configurationService` and try to modify Sendgrid (or Slack, or Klaviyo...)

Reply:
Renamed to be more specific :+1: